# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../openl2tp/openl2tp-ppp_setup_timer_with_norpc.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

When building without L2TP_FEATURE_RPC_MANAGEMENT ppp setup timer, that is
created in plugins/ppp_unix.c:1222, always expire. The only code path that
deactivates it is triggered by openl2tp.so pppd plugin hook. This plugging
doesn't seem to contain any RPC code now and use unix socket to communicate
with opel2tpd and trigger l2tp_session_ppp_updown_ind that calls
ppp_unix_session_ppp_created_hook to stop ppp setup timer. Uncommenting this
plugin loading makes non-rpc version usable.

Patch URL: http://sourceforge.net/tracker/download.php?group_id=118353&atid=680935&file_id=397349&aid=3149171

Patch Reference: http://sourceforge.net/tracker/?func=detail&aid=3149171&group_id=118353&atid=680935

 plugins/ppp_unix.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/plugins/ppp_unix.c b/plugins/ppp_unix.c
index 869066f..6e1b60b 100644
--- a/plugins/ppp_unix.c
+++ b/plugins/ppp_unix.c
@@ -892,11 +892,9 @@ static int ppp_unix_pppd_spawn(struct ppp_context *ppp)
 
 		argv[arg++] = "plugin";
 		argv[arg++] = "pppol2tp.so";
-#ifdef L2TP_FEATURE_RPC_MANAGEMENT
 		/* This plugin requires pppol2tp.so to be loaded first */
 		argv[arg++] = "plugin";
 		argv[arg++] = "openl2tp.so";
-#endif
 
 		if (!ppp_unix_use_dummy_pppd()) {
 			int flags;
